src/CatBox.NET/Client/CatBox/CatBoxClient.cs

```diff
@@ -0,0 +1,238 @@
+﻿using System.Runtime.CompilerServices;
+using CatBox.NET.Enums;
+using CatBox.NET.Requests.CatBox;
+using Microsoft.Extensions.Options;
+using static CatBox.NET.Client.Common;
+
+namespace CatBox.NET.Client;
+
+public class CatBoxClient : ICatBoxClient
+{
+    private readonly HttpClient _client;
+    private readonly CatboxOptions _catboxOptions;
+
+    /// <summary>
+    /// Creates a new <see cref="CatBoxClient"/>
+    /// </summary>
+    /// <param name="client"><see cref="HttpClient"/></param>
+    /// <param name="catboxOptions"><see cref="IOptions{TOptions}"/></param>
+    /// <exception cref="ArgumentNullException"></exception>
+    public CatBoxClient(HttpClient client, IOptions<CatboxOptions> catboxOptions)
+    {
+        _client = client ?? throw new ArgumentNullException(nameof(client), "HttpClient cannot be null");
```

Comment:
Use `Throw.IfNull` to be more consistent since there's no real special comment here.

Reply:
Done